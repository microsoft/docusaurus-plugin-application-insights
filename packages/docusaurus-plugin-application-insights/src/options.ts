export type ApplicationInsightsOptions = {
  instrumentationKey: string;
};

export type Options = Partial<ApplicationInsightsOptions>;
