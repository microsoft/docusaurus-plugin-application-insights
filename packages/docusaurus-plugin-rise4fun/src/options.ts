import type { ApplicationInsightsOptions } from "docusaurus-plugin-application-insights";
import type { PluginOptions as CompileCodePluginOptions } from "docusaurus-remark-plugin-compile-code";
export type PluginOptions = {
    appInsights?: ApplicationInsightsOptions;
    compileCode?: CompileCodePluginOptions;
};

export type Options = Partial<PluginOptions>;
