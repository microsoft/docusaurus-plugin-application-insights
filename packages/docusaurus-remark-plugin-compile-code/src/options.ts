export interface LangOptions {
    lang: string;
    version?: string;
    command?: string;
    args?: string[];
    timeout?: number;
    extension?: string;
    outputLang?: string;
    outputMeta?: string;

    nodeBin?: string;
    npmPackage?: string;
}

export type PluginOptions = {
    outputPath?: string;
    langs: LangOptions[];
    concurrency?: number;
};
