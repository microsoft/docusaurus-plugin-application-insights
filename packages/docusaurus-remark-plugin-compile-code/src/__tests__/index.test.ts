/**
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

import path from "path";
import vfile from "to-vfile";
import mdx from "remark-mdx";
import remark from "remark";
import plugin from "../index";
import { PluginOptions } from "../options";

const processFixture = async (name: string, options: PluginOptions) => {
    const filePath = path.join(__dirname, "__fixtures__", `${name}.md`);
    const file = await vfile.read(filePath);
    const result = await remark().use(mdx).use(plugin, options).process(file);

    return result.toString();
};

const options: PluginOptions = {
    langs: [
        {
            lang: "ts",
            version: "0.0.0",
            nodeBin: "tsc",
            npmPackage: "typescript",
        },
    ],
};

describe("extract-code plugin", () => {
    it("works on snippet file", async () => {
        const result = await processFixture("snippets", options);
        expect(result).toMatchSnapshot();
    });
});
