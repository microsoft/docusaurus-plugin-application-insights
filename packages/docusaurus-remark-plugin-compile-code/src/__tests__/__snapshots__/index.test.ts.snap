// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`extract-code plugin works on compile file 1`] = `
"Some snippets

\`\`\`lisp
foo
\`\`\`

    prefix

    foo
"
`;

exports[`extract-code plugin works on meta file 1`] = `
"Testing meta

\`\`\`fail skip foo
don't run this
\`\`\`

next, broken

\`\`\`fail ignore-errors foo
ignore errors
\`\`\`

    Error: fail

missing lang meta

\`\`\`fail
fail
\`\`\`

prefix

\`\`\`lisp prefix

\`\`\`

    prefix

\`\`\`lisp prefix
prefix
\`\`\`

    prefix
"
`;

exports[`extract-code plugin works on nodebin file 1`] = `
"Some snippets

\`\`\`ts
let x = 0
\`\`\`

next, broken

\`\`\`ts ignore-errors
sadfj asd fasdf
\`\`\`

    input.ts(1,1): error TS1434: Unexpected keyword or identifier.
    input.ts(1,7): error TS1434: Unexpected keyword or identifier.
"
`;
